analyzers/src/SonarAnalyzer.CSharp/Rules/ParameterValidationInAsyncShouldBeWrapped.cs

```diff
@@ -42,13 +42,13 @@ public sealed class ParameterValidationInAsyncShouldBeWrapped : SonarDiagnosticA
             DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
+        protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
                     var methodDeclaration = (MethodDeclarationSyntax)c.Node;
-                    if (!methodDeclaration.Modifiers.Any(SyntaxKind.AsyncKeyword))
+                    if (!methodDeclaration.Modifiers.Any(SyntaxKind.AsyncKeyword)
+                        || c.SemanticModel.GetTypeInfo(methodDeclaration.ReturnType).Type.SpecialType == SpecialType.System_Void)
```

Comment:
Better API
```suggestion
                        || c.SemanticModel.GetDeclaredSymbol(methodDeclaration).ReturnsVoid)
```

Reply:
I avoided on purpose getting the symbol, because that can trigger a compilation; whereas I only need the type info for this purpose. 

I will try and do a small benchmark

Reply:
From a peek to Roslyn, it uses `GetSemanticInfoSymbolInNonMemberContext` anyway

Reply:
I changed it to remove SemanticModel